dsl/camel-kamelet-main/src/main/java/org/apache/camel/main/download/SagaDownloader.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.main.download;
+
+import java.util.Arrays;
+
+import org.apache.camel.NamedNode;
+import org.apache.camel.Processor;
+import org.apache.camel.Route;
+import org.apache.camel.main.KameletMain;
+import org.apache.camel.model.SagaDefinition;
+import org.apache.camel.reifier.ProcessReifier;
+import org.apache.camel.reifier.ProcessorReifier;
+import org.apache.camel.saga.InMemorySagaService;
+
+/**
+ * When using saga then we need to register a CamelSagaService
+ */
+public class SagaDownloader {
+
+    private SagaDownloader() {
+    }
+
+    public static void registerDownloadReifiers(KameletMain main) {
+
+        ProcessorReifier.registerReifier(SagaDefinition.class,
+                (route, processorDefinition) -> {
+                    main.bind("inMemorySagaService", new InMemorySagaService());
```

Comment:
You need to download camel-saga also like transacted do

Reply:
Well, `camel-saga` is a camel componet which can commit or rollback the saga transaction manually. see https://camel.apache.org/components/4.8.x/saga-component.html and it can be detected in
```java
to("saga:complete")
```

But here is the dsl like
```java
from("direct:saga")
    .saga()
    .compensation(...)
    .to(...)
```
I think they are different and no need to download `camel-saga` here.

Reply:
the point is that end users dont know about all that stuff - they add saga and then need jbang to help download camel-saga and camel-lra to make it useable. the in memory is only for quick testing.

Reply:
Ok, I get it.

Reply:
@davsclaus I added `camel-saga` and `camel-lra` to the downloader.